java/arcs/sdk/EdgeHandles.kt

```diff
@@ -0,0 +1,161 @@
+package arcs.sdk
+
+import arcs.core.entity.Handle
+import arcs.core.entity.ReadCollectionHandle
+import arcs.core.entity.ReadSingletonHandle
+import arcs.core.entity.Storable
+import arcs.core.entity.WriteCollectionHandle
+import arcs.core.entity.WriteSingletonHandle
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.withContext
+
+/*
+ * To facilitate safe read/write access to data handles in running arcs, particles can be
+ * annotated with '@edge' in the manifest. This will cause the following set of changes to
+ * the code generation output:
+ *  - The generated class will be concrete and final instead of an abstract base class.
+ *  - The particle lifecycle will be managed internally to this class; clients will not have
+ *    any access to lifecycle events.
+ *  - Handles will be exposed via the Edge*Handle interfaces given below. All edge handle methods
+ *    are asynchronous and may be invoked at any time after the particle has been created.
+ *  - All read operations will return a CompletableDeferred around the requested value.
```

Comment:
`would it be easier for the caller if we just suspend until the result is available?` -> I thought a Deferred allows more flexibility. It's possible that callers may be able to continue other processing before accessing the result of a read op. I guess that could be over-engineering it.

`you can store the pending ops as continuation instead than deferred, and complete them when ready` -> I though that's what Deferreds were for? How do you store a suspend call (without requiring the caller to use `launch`-style ops)?

Reply:
I don't think a Deferred allows more flexibility. If you don't want to wait for the read (which I think would be the less common use case) you can always launch { handle.fetch() }.
I think you can store the continuation for example like this: 
https://github.com/PolymerLabs/arcs/blob/2fb0c04835c7f7310707364d78f164fa8dadf4fd/java/arcs/sdk/android/labs/host/IntentHostAdapter.kt#L46

Reply:
Ok, I'll just make the returns direct.

That example shows a continuation being passed in, not how to create one, but I gather you're referring to using the `launch` approach (which I was actively trying to avoid requiring on the caller, but I suspect you're right that it's an unlikely use case).

Reply:
you can create one like this:
https://github.com/PolymerLabs/arcs/blob/2fb0c04835c7f7310707364d78f164fa8dadf4fd/java/arcs/sdk/android/labs/host/IntentHostAdapter.kt#L82